### ISSUE TITLE
IDISP003 should not warn when if returning in loop

### ISSUE THREAD

**Issue by JohanLarsson** [DotNetAnalyzers/IDisposableAnalyzers#137]
```cs
private static bool TryGetStream(string fileName, Assembly callingAssembly, out Stream result)
{
    foreach (var name in callingAssembly.GetManifestResourceNames())
    {
        if (name.EndsWith(fileName, ignoreCase: true, culture: CultureInfo.InvariantCulture))
        {
            result = callingAssembly.GetManifestResourceStream(name);
            return true;
        }
    }

    result = null;
    return false;
}
```
